service/history/queuev2/virtual_slice.go

```diff
@@ -316,51 +353,48 @@ func mergeVirtualSlicesByRange(left, right *virtualSliceImpl) VirtualSlice {
 	}
 }
 
-func mergeGetTaskProgress(left, right []*GetTaskProgress) []*GetTaskProgress {
+func mergeGetTaskProgressWithSamePredicate(left, right []*GetTaskProgress) []*GetTaskProgress {
 	mergedProgress := []*GetTaskProgress{}
 	leftIndex := 0
 	rightIndex := 0
 	for leftIndex < len(left) && rightIndex < len(right) {
-		if left[leftIndex].NextTaskKey.Compare(right[rightIndex].NextTaskKey) <= 0 {
-			mergedProgress = appendOrMergeProgress(mergedProgress, left[leftIndex])
+		if left[leftIndex].Range.InclusiveMinTaskKey.Compare(right[rightIndex].Range.InclusiveMinTaskKey) <= 0 {
```

Comment:
bug fix, unit test didn't fail because mergeProgressWithSamePredicate function swap left and right